client/main.lua

```diff
@@ -70,168 +70,155 @@ local function registerDutyTarget()
+            end,
+            onExit = function()
+                lib.hideTextUI()
+                isInsideStashZone = false
             end
-
-            isInsideStashZone = isPointInside
-        end)
-
+        })
         Config.Targets[stashTargetBoxId] = {created = true, zone = zone}
     end
 end
 
 local function registerGarageZone()
     local coords = Config.Locations.vehicle
-    local vehicleZone = BoxZone:Create(coords.xyz, 5, 15, {
-        name = 'vehicleZone',
-        heading = 340.0,
-        minZ = coords.z - 1.0,
-        maxZ = coords.z + 5.0,
-        debugPoly = false
-    })
-
-    vehicleZone:onPlayerInOut(function (isPointInside)
-        if isPointInside and QBX.PlayerData.job.onduty then
-            local inVehicle = cache.vehicle
-            if inVehicle then
-                lib.showTextUI(Lang:t('labels.h_vehicle'), {
-                    position = 'left'
-                })
-            else
-                lib.showTextUI(Lang:t('labels.g_vehicle'), {
-                    position = 'left'
-                })
+    local size = Config.LocationsSize.vehicle
+
+    lib.zones.box({
+        coords = coords,
+        size = size,
+        rotation = 339.75,
+        debug = Config.DebugPoly,
+        onEnter = function()
+            if QBX.PlayerData.job.onduty then
+               -- local inVehicle = cache.vehicle
+                 if cache.vehicle then
+                    lib.showTextUI(Lang:t('labels.h_vehicle'), { position = 'left-center' })
+                 else
+                    lib.showTextUI(Lang:t('labels.g_vehicle'), { position = 'left-center' })
+                 end
+                 isInsideGarageZone = true
             end
-        else
-            lib.hideTextU()
+        end,
+        onExit = function()
+            lib.hideTextUI()
+            isInsideGarageZone = false
         end
-
-        isInsideGarageZone = isPointInside
-    end)
+    })
 end
 
+
 local function destroyVehiclePlateZone(id)
     if plateZones[id] then
-        plateZones[id]:destroy()
+        plateZones[id]:remove()
         plateZones[id] = nil
     end
 end
 
 local function registerVehiclePlateZone(id, plate)
     local coords = plate.coords
     local boxData = plate.boxData
-    local plateZone = BoxZone:Create(coords.xyz, boxData.length, boxData.width, {
-        name = plateTargetBoxId .. id,
-        heading = boxData.heading,
-        minZ = coords.z - 1.0,
-        maxZ = coords.z + 3.0,
-        debugPoly = boxData.debugPoly
-    })
-
-    plateZones[id] = plateZone
 
-    plateZone:onPlayerInOut(function (isPointInside)
-        if isPointInside and QBX.PlayerData.job.onduty then
-            if plate.AttachedVehicle then
+    local plateZone = lib.zones.box({
+        coords = coords,
+        size = boxData.size,
+        rotation = boxData.rotation,
+        debug = Config.DebugPoly,
+        onEnter = function()
+            if QBX.PlayerData.job.onduty then
```

Comment:
good opportunity here for a guard clause to gate the unduly check https://manason.github.io/effective-fivem-lua/functions/#avoid-nesting